internal/broker/broker.go

```diff
@@ -311,35 +326,45 @@ func (b *Broker) generateUILayout(session *sessionInfo, authModeID string) (map[
 
 	var uiLayout map[string]string
 	switch authModeID {
-	case "device_auth":
+	case authmodes.DeviceID, authmodes.DeviceQrID:
 		ctx, cancel := context.WithTimeout(context.Background(), maxRequestDuration)
 		defer cancel()
 		response, err := session.authCfg.oauth.DeviceAuth(ctx)
 		if err != nil {
-			return nil, fmt.Errorf("could not generate QR code layout: %v", err)
+			return nil, fmt.Errorf("could not generate Device Authentication code layout: %v", err)
 		}
 		session.authInfo["response"] = response
 
-		uiLayout = map[string]string{
-			"type": "qrcode",
-			"label": fmt.Sprintf(
+		var label string
+		if authModeID == authmodes.DeviceQrID {
+			label = fmt.Sprintf(
 				"Scan the QR code or access %q and use the provided login code",
 				response.VerificationURI,
-			),
+			)
+		} else {
+			label = fmt.Sprintf(
+				"Access %q and use the provided login code",
+				response.VerificationURI,
+			)
+		}
```

Comment:
I think it's best to assume the "default" label as being without the code here and then assign the other one in case QRCode is supported. This way we can drop the else 🙃

Reply:
Hehe, yeah, I was initially doing this, but then I decided not to follow the happy-path as there was really no point to allocate things if not needed...

I can move this to a function though :)